strapi/strapi
# fix: asset url appending

### What does it do?

This fix first checks if an asset URL already contains a `?` before trying to append one. If one is already present, it appends using an `&` instead.

I haven't really worked in the Strapi code base before, so I'm not sure how safe it is to call `thumbnail.includes()` without any check to confirm that thumbnail is defined and a string.

### Why is it needed?

The current implementation will append `?` to a URL that already has a `?` in it, creating an invalid URL and causing the browser to fail to fetch the asset. For me, this was a problem while using the AWS S3 provider's signed URLs feature, which already appends GET parameters to the asset URL.

### How to test it?

This can be tested with a Strapi setup that uses the Strapi AWS S3 provider with signed URLs enabled. At least one image must be uploaded to the bucket so that it is attempted to be loaded when navigating to the Media Library page of the Strapi admin area.

Without the fix present, the image will try to be fetched with a URL such as `
https://my-bucket.s3.us-west-2.amazonaws.com/thumbnail_image1_d87bdf0614.jpg?AWSAccessKeyId=111&Expires=1693360000&Signature=1%3D?2023-05-19T03:00:25.625Z`. This will return a 403 and a broken image will appear on the page. Removing the `?2023-05-19T03:00:25.625Z` and navigating to the URL should successfully fetch the image.

With the fix present, the image URL will end with `&2023-05-19T03:00:25.625Z` which will allow it to be fetched successfully and displayed on the page.

### Related issue(s)/PR(s)

* Fixes https://github.com/strapi/strapi/issues/17655
